src/renderers/dom/client/eventPlugins/ChangeEventPlugin.js

```diff
@@ -45,6 +47,35 @@ var eventTypes = {
   },
 };
 
+function getInstIfValueChanged(targetInst, target) {
+  var value = getInputValue(target);
+  var lastValue = target[LAST_VALUE_KEY];
+
+  if (!target.hasOwnProperty(LAST_VALUE_KEY) || value !== lastValue) {
+    target[LAST_VALUE_KEY] = value;
```

Comment:
changing a value on on uncontrolled input should trigger an `onChange` which will trigger an update here (in text inputs anyway) since the code listens for both onChange and onInput. Missing an update should be fine since the next raised event will update the cached value 

That being said is that what we want in the first place? native `onInput` doesn't trigger when you set `node.value`; the IE8 pollyfill even wraps the `value` descriptor in order to detect and silence value changes triggered from javascript. If we need that consistenty (which makes sense to me) then wrapping the value descriptor in the same way would allow for that, but I'm not sure if that causes any significant perf regressions, getting and setting all value descriptors on listened to inputs.